graviti/manager/dataset.py

```diff
@@ -41,12 +37,44 @@ class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-att
         dataset_id: Dataset ID.
         name: The name of the dataset, unique for a user.
         alias: Dataset alias.
+        default_branch: The default branch of dataset.
         commit_id: The commit ID.
-        branch_name: The branch name.
+        created_at: The time when the dataset is created.
+        updated_at: The time of last update.
+        owner: The owner of the dataset.
+        is_public: Whether the dataset is public.
+        config: The config of dataset.
+        branch: The current branch name of the dataset.
 
     """
 
     _data: Dict[str, DataFrame]
+    _T = TypeVar("_T", bound="Dataset")
+    _repr_attrs: Tuple[str, ...] = (
+        "dataset_id",
+        "alias",
+        "default_branch",
+        "commit_id",
+        "created_at",
+        "updated_at",
+        "is_public",
+        "config",
+        "branch",
+    )
+
+    _access_key: str = attr(key="access_key")
+    _url: str = attr(key="url")
+    _dataset_id: str = attr(key="id")
+    _name: str = attr(key="name")
+    _alias: str = attr(key="alias")
+    _default_branch: str = attr(key="default_branch")
+    _commit_id: str = attr(key="commit_id")
+    created_at: str = attr()
+    updated_at: str = attr()
+    _owner: str = attr(key="owner")
+    _is_public: bool = attr(key="is_public")
+    _config: str = attr(key="config")
+    _branch: Optional[str] = attr(key="branch")
```

Comment:
why branch is optional？

Reply:
when we checkout to a commit, the branch will be None

Reply:
got it